move code to where it is used

jalvz opened the pull request:
## What does this PR do?

This moves helper functions and variables closer to where they are generally used.

## Why is it important?

It follows standard Go practices and aids IDE auto-completion. Normally, you would not call a function defined in a file `foo_test.go` from `foo.go`; usually is the other way around. Godog compiles fine by linking the right files, but it confuses the hell out of IDEs and me too ;) 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally

## How to test this PR locally

Run tests

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp%2FPR-859/detail/PR-859/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp%2FPR-859/detail/PR-859/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp%2FPR-859/detail/PR-859/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp%2FPR-859/detail/PR-859/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://e2e-testing_859.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats


* Build Cause: `Pull request #859 opened`



* Start Time: 2021-03-08T14:12:59.181+0000


* Duration: 25 min 43 sec


* Commit: c4523c1c55b9b3f225ccd22c7f9f6598d1d5c37f



#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 129  |
| Skipped      | 0 |
| Total        | 129   |


#### Trends :test_tube:


![Image of Build Times](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-mbp/job/PR-859/buildTimeGraph/png)


![Image of Tests](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-mbp/job/PR-859/1/testReport/history/countGraph/png?start=0&amp;end=25)


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->












## :green_heart: Flaky test report
Tests succeeded.


<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

### Test stats :test_tube:
| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 129  |
| Skipped      | 0 |
| Total        | 129   |



</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->

mdelapenya commented:
I've been thinking about this for a long time and never gave it a try, but you make a good point about file names. How would you see removing the `ingest_manager_test.go` to `ingest_manager.go`? The old name was there for historical not-valid-anymore reasons, so I'd see with good eyes to rename the file. In fact, newer code is using the approach you brightly points: using `_test.go` for unit tests.

I'd prefer renaming the file because the ingest_manager file represents the main entrypoint for the test suite: it defines main variables, and is the responsible of creating the downstream test suite structs (fleet and standalone).

Wdyt?

jalvz commented:
@mdelapenya that sounds good to me. However I tried and it didn't work, it seems that godog will only search for the entry point in a `_test.go` file. So following your suggestion as much as I could, I moved everything to `ingest_manager.go` and kept the bare minimum in the test file. 
Additionally, I added a `TestMain` function to integrate `godog` with `go test`, meaning that bdd tests can be run with `go test` as well. We don't need to change how it is run, but just as an alternative - eg. this should make possible to run and debug tests from the IDE even if the IDE doesn't support godog.   

I know there is a merge conflict, but lmk what do you think generally.

mdelapenya commented on e2e/_suites/fleet/ingest_manager_test.go:
```diff
@@ -0,0 +1,43 @@
+	initializeIngestManagerTestScenario(ctx)
+}
+
+func TestMain(m *testing.M) {
```
Although I like this approach, I think it's mixing the unit tests from the godog tests.

There is a unit test layer (`make -C e2e unit-test`) that executes only unit tests, not BDD/godog tests, and it seems this change is adding the godog tests to the execution of the unit tests, which uses `[gotestsum](https://github.com/gotestyourself/gotestsum)` as test runner (uses go test plus pigments)

jalvz commented:
Superseded by #865
